Housekeeping changes for Knife (#14831)
chef/chef: docs/dev/how_to/releasing_chef_infra.md

Please make sure cookstyle is working properly & auto correcting detected offenses for any of the cookbooks you are trying to test against the newer version of Chef Infra Client
Reference doc - https://github.com/chef/cookstyle#usage

> ### Bumping Chef version inside Knife
> Knife is released as a separate gem but currently resides under same github repository.
> Whenever we have released a new version of Infra Client make sure to bump the same in knife's Gemfile.lock.
> Create PR after running following commands and merge the same.
> `cd knife`
> `bundle update chef --conservative`
> 
> ### Relax

Reviewer: ```suggestion
### Bumping Chef version inside Knife

Knife is released as a separate gem, but currently resides under the same GitHub repository.
Whenever we release a new Infra Client version, bump the version in knife's Gemfile.lock.

Create PR and merge a PR after running the following commands:

- `cd knife`
- `bundle update chef --conservative`

### Relax
```

Author: done
